packages/product/integration-tests/__tests__/services/product-module-service/product-variants.spec.ts

```diff
@@ -0,0 +1,172 @@
+import { initialize } from "../../../../src"
+import { DB_URL, TestDatabase } from "../../../utils"
+import { IProductModuleService } from "@medusajs/types"
+import { Product, ProductVariant } from "@models"
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+import { ProductTypes } from "@medusajs/types"
+
+describe("ProductModuleService product variants", () => {
+  let service: IProductModuleService
+  let testManager: SqlEntityManager
+  let repositoryManager: SqlEntityManager
+  let variantOne: ProductVariant
+  let variantTwo: ProductVariant
+  let productOne: Product
+  let productTwo: Product
+
+  beforeEach(async () => {
+    await TestDatabase.setupDatabase()
+    repositoryManager = await TestDatabase.forkManager()
+
+    service = await initialize({
+      database: {
+        clientUrl: DB_URL,
+        schema: process.env.MEDUSA_PRODUCT_DB_SCHEMA,
+      },
+    })
+
+    testManager = await TestDatabase.forkManager()
+
+    productOne = testManager.create(Product, {
+      id: "product-1",
+      title: "product 1",
+      status: ProductTypes.ProductStatus.PUBLISHED,
+    })
+
+    productTwo = testManager.create(Product, {
+      id: "product-2",
+      title: "product 2",
+      status: ProductTypes.ProductStatus.PUBLISHED,
+    })
+
+    variantOne = testManager.create(ProductVariant, {
+      id: "test-1",
+      title: "variant 1",
+      inventory_quantity: 10,
+      product: productOne,
+    })
+
+    variantTwo = testManager.create(ProductVariant, {
+      id: "test-2",
+      title: "variant",
+      inventory_quantity: 10,
+      product: productTwo,
+    })
+
+    await testManager.persistAndFlush([variantOne, variantTwo])
+  })
+
+  afterEach(async () => {
+    await TestDatabase.clearDatabase()
+  })
+
+  describe("listAndCountVariants", () => {
+    it("should return variants and count queried by ID", async () => {
+      const results = await service.listAndCountVariants({
+        id: variantOne.id,
+      })
+
+      expect(results[1]).toEqual(1)
+      expect(results[0]).toEqual([
+        expect.objectContaining({
+          id: variantOne.id,
+          title: "variant 1",
+          inventory_quantity: "10",
+        }),
+      ])
+    })
+
+    // TODO: investigate why pagination is incorrect
+    it.skip("should return variants and count based on the 'take' parameter", async () => {
```

Comment:
question: When will we attend to this todo and the skipping of the test?

Reply:
I've already begun investigating this, this should be in the next PR.

Reply:
Didn't add it here since it's a general issue with pagination across services.